plugins/swf-backend/src/service/router.ts

```diff
@@ -125,9 +125,23 @@ function setupInternalRoutes(
     const {
       params: { swfId },
     } = req;
-    const wsRequest = await fetch(
-      `${kogitoBaseUrl}:${kogitoPort}/management/processes/${swfId}/source`,
-    );
+
+    let wsRequest;
+    let errorCount = 0;
+    // execute with retry
+    while (errorCount < 10) {
```

Comment:
Your loop is much nicer than [mine](https://github.com/manstis/backstage/blob/b74a1ca7d95673b1b28a22959a265ccdab6fc451/plugins/swf-backend/src/service/router.ts#L287-L309).

Would you mind changing mine too?